iati/core/tests/test_data.py

```diff
@@ -406,3 +408,112 @@ def test_dataset_xml_str_source_around_line_invalid_context_lines(self, invalid_
         for line_num in range(0, num_lines_xml):
             with pytest.raises(TypeError):
                 data.source_around_line(line_num, invalid_value)
+
+
+class TestDatasetVersionDetection(object):
+    """A container for tests relating to detecting the version of a Dataset."""
+
+    @pytest.fixture(params=[
+        ('iati-activities', 'iati-activity'),
+        ('iati-organisations', 'iati-organisation')
+    ])
+    def iati_tag_names(self, request):
+        """Return the tag names for an activity or organisaion dataset."""
+        output = collections.namedtuple('output', 'root_element child_element')
+        return output(root_element=request.param[0], child_element=request.param[1])
+
+    @pytest.mark.parametrize("version", iati.core.utilities.versions_for_integer(1))
+    def test_detect_version_v1_simple(self, iati_tag_names, version):
+        """Check that a version 1 dataset is detected correctly.
+        Also checks that version numbers containing whitespace do not affect version detection.
+        """
+        data = iati.core.Dataset("""
+        <{0} version="{1}">
+            <{2} version="{1}"></{2}>
+            <{2} version="{1}  "></{2}>
+            <{2} version="   {1}"></{2}>
+            <{2} version="   {1}   "></{2}>
+        </{0}>
+        """.format(iati_tag_names.root_element, version, iati_tag_names.child_element))
```

Comment:
Since `version` is optional, while the others exist for all Datasets, suggest re-ordering so that `version` is `{2}`, allowing the others to be consistent.

(noted here, there may be others)